### DISCUSSION TITLE
Problem when use the text-Enhancement

### DISCUSSION THREAD

**Discussion by DFanny-5** [snakers4/silero-models#180] [Q&A]
I find a problem in the text-enhancement's output. May be it is not an issue but I hope I can get some help to solve it. Currently I am trying to perform the re-punctuation on the out come of my other ASR model.

Input: 
in the next panel you will meet cathy n davidson founding director of the futures initiative mary beard professor of classics of the university of cambridge martin professor of integrative medicine at karolinska institutet and kilo nobel prize laureate in physics in two thousand nineteen the topic is the future of higher education and moderator is adam smith thank you to the nobel prize committee for hosting this forum on the challenges of education and thank you to all of you in the audience who are taking time out from your very limited lives to be here to day that means you care and i think there's no subject more important to care about right now than education economist mariana moscato has said we should be thinking ahead to twenty twenty three i**n other words hopefully the time when the pandemic is passed away thinking about what it means to return to normal**
Output:
In the next panel you will meet Cathy n Davidson, founding director of the Futures Initiative, Mary Beard, Professor of Classics of the University of Cambridge, Martin, Professor of Integrative medicine at Karolinska Institutet and Kilo Nobel Prize Laureate in Physics in two thousand nineteen. The topic is the future of higher education and moderator is Adam Smith. Thank you to the Nobel Prize Committee for hosting this forum on the challenges of education and thank you to all of you in the audience who are taking time out from your very limited lives to be here to day that means you care and i think there's no subject more important to care about right now than education. Economist Mariana Moscato has said we should be thinking ahead to twenty twenty-three. **In In other words, hopefully, the time when the pandemic is passed away, thinking about what it means to return to normal.**

As you can see that an additive 'In' was added to the result in the last sentence. I wonder why this would happen because the re-punctuation seems should not add any additive word to the input, isn't it? May I ask how can i solve it or where i may find the source code of how the capitalized word was added in this model? Many thanks

**Comment by snakers4** [comment]
Looks like a bug, you can just seep through the output and compare it word by word omitting extra words.

**Reply by DFanny-5** [reply]
Thanks for replying! I may try that. Yes, this seems like a bug. If I remove one of the "twenty" from the previous "twenty twenty three" or reduce the size of the unpunctuated text to a shorter length the problem seems got solved. I hope these may help you or people with the same question in the future to solve this problem.

For the word's comparison and omission, is there any algorithm you may suggest such as the Hamming algorithm which I know was used for communication, I just want to make this new feature fancy and effective. Many thanks

**Reply by snakers4** [reply]
> or reduce the size of the unpunctuated text to a shorter length the problem seems got solved

This is also a good idea.
But doing some rudimentary checks to ensure no hallucinations also makes sense.

**Reply by DFanny-5** [reply]
> > or reduce the size of the unpunctuated text to a shorter length the problem seems got solved
> 
> This is also a good idea. But doing some rudimentary checks to ensure no hallucinations also makes sense.

I have tried to reduce the size but I think it is hard to guarantee the error will not happen again. Is there any suggestion I can get from you about the techniques can be used for the rudimentary checks. Any suggestion will be helpful, I can search for the paper and study on it if you just give me a terminology.
Many thanks
